recipes/szip/all/conanfile.py

```diff
@@ -28,72 +31,69 @@ class SzipConan(ConanFile):
         "enable_large_file": True,
     }
 
-    generators = "cmake"
-    _cmake = None
-
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    @property
-    def _build_subfolder(self):
-        return "build_subfolder"
-
     def export_sources(self):
-        self.copy("CMakeLists.txt")
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            self.copy(patch["patch_file"])
+        export_conandata_patches(self)
 
     def config_options(self):
         if self.settings.os == "Windows":
             del self.options.fPIC
 
     def configure(self):
         if self.options.shared:
-            del self.options.fPIC
-        del self.settings.compiler.libcxx
-        del self.settings.compiler.cppstd
+            try:
+                del self.options.fPIC
+            except Exception:
+                pass
+        try:
+            del self.settings.compiler.libcxx
+        except Exception:
+            pass
+        try:
+            del self.settings.compiler.cppstd
+        except Exception:
+            pass
+
+    def layout(self):
+        cmake_layout(self, src_folder="src")
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+        get(self, **self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
 
     def build(self):
-        for patch in self.conan_data.get("patches", {}).get(self.version, []):
-            tools.patch(**patch)
-        tools.replace_in_file(os.path.join(self._source_subfolder, "CMakeLists.txt"),
+        apply_conandata_patches(self)
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
                               "set (CMAKE_POSITION_INDEPENDENT_CODE ON)", "")
-        cmake = self._configure_cmake()
+        replace_in_file(self, os.path.join(self.source_folder, "CMakeLists.txt"),
+                              "option (BUILD_SHARED_LIBS \"Build Shared Libraries\" ON)", "")
```

Comment:
You are adding weird things, this should be set by the toolchain and it's not force cache so I am really surprised you need this

Maybe try this secret

https://github.com/conan-io/conan-center-index/blob/843317aed0272023e4be86eb6616c645f8f9ff8e/recipes/libcorrect/all/conanfile.py#L61-L62

Reply:
Look at the build logs there's actually a warning saying to do this, so thats actually expected... TIL

Reply:
Thanks, I was not enjoying adding those lines